HW0/P4/P4.py

```diff
@@ -30,47 +32,114 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
+    
+    TruTraj = np.loadtxt("P4_trajectory.txt",dtype='float',delimiter=',')
+    xTru_coords, yTru_coords, zTru_coords = TruTraj[:,0], TruTraj[:,1], TruTraj[:,2]
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    #Plotting the true position
+    ax.plot(xTru_coords, yTru_coords, zTru_coords, '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    
+    ObsTraj = np.loadtxt("P4_measurements.txt",dtype='float',delimiter=',')
+    xObs_coords, yObs_coords, zObs_coords = ObsTraj[:,0], ObsTraj[:,1], ObsTraj[:,2]
+    
+    #Plotting the measured position
+    #ax.plot(xObs_coords, yObs_coords, zObs_coords,'.g', label='Observed trajectory')
+    
+    xtld = ObsTraj.dot(np.diag([1/rx, 1/ry, 1/rz])) # I decided to try and streamline this...
+                                           #  which may be a sign that I'm getting comfy
+    #Plotting the approximate position
+    ax.plot(xtld[:,0],xtld[:,1],xtld[:,2],'.r', label='Approximate Position')                                        
+    
+    
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
+	#Propagation Matrix
+    A = np.diag([1, 1, 1, 1-c*dt, 1-c*dt, 1-c*dt]) + np.diag([dt, dt, dt],3)
+    
+    #Accounting for gravity in update
+    a = np.zeros((A.shape[0])) 
+    a[-1] = g*dt
+    a = a.reshape([6,1])
     # s = ?
 
     # Initial conditions for s0
+    s0 = np.matrix(TruTraj[0,:]).T
+    
     # Compute the rest of sk using Eq (1)
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+    sK = np.matrix(np.zeros((6,np.amax(TruTraj.shape)))) #Initializing array for state predictions
+    sK[:,0] = s0                              #Placing initial measurement as first column
+    
+    for i in range(1,np.amax(TruTraj.shape)):
+         sK[:,i] = A * sK[:,i-1] + a
+    
+    sK = np.asarray(sK) # Converting state prediction matrix to an array (for plotting??)
```

Comment:
An alternative is to start off with sK being an np.array (rather than np.matrix) and then use np.dot to do matrix multiplication, but this looks like a perfectly good way too.